src/main/java/executor/command/CommandGetSpendingByYear.java

```diff
@@ -24,10 +27,37 @@ public CommandGetSpendingByYear(String userInput) {
     public void execute(Wallet wallet) {
         ReceiptTracker receiptsInYear = new ReceiptTracker();
         String yearStr = Parser.parseForPrimaryInput(CommandType.EXPENDEDYEAR, userInput);
-        int year = Integer.parseInt(yearStr);
-        receiptsInYear = wallet.getReceipts().findReceiptByYear(year);
-        Double totalMoney = receiptsInYear.getTotalCashSpent();
-        Ui.dukeSays("The total amount of money spent in " + year + " : " + totalMoney);
+
+        if (yearStr.isEmpty()) {
+            Ui.dukeSays("No year input detected. FORMAT : expendedyear 2019");
+            return;
+        }
+
+        if (yearStr.length() != 4) {
+            Ui.dukeSays("Year input contains extra number of variables. FORMAT : expendedyear 2019");
+            return;
+        }
+
```

Comment:
These lines are outside the try block. They should be inside no?

Reply:
These parts don't throw any exceptions. The parts inside the try block tends to do so.